test/boost/sstable_resharding_test.cc

```diff
@@ -81,7 +81,10 @@ void run_sstable_resharding_test() {
     auto filter_fname = sstables::test(sst).filename(component_type::Filter);
     uint64_t bloom_filter_size_before = file_size(filter_fname).get0();
 
-    auto creator = [&env, &cf, &tmp, version] (shard_id shard) mutable {
+
+    auto descriptor = sstables::compaction_descriptor({sst}, 0, std::numeric_limits<uint64_t>::max());
+    descriptor.options = sstables::compaction_options::make_reshard();
```

Comment:
How about adding `options` to `compaction_descriptor` constructor, along with `sstables`, `level`, and `max_sstable_bytes`? (and default to `make_regular` as it is now)?
It looks like it could come handy in several places and it can save initializing options and then overwriting it.

Reply:
I considered, the main problem is that there are many callers we'd have to patch. I am not sure it is worth the effort.

Reply:
I tried and indeed there are many sites. @bhalevy I don't dislike the idea.
But how about we do this in a follow up patch? It's technically not related to what I am trying to do here. I am happy to clean that up separately but I'd like to get those in

Reply:
Just realized last argument of c__d's constructor is c__o default'd to regular, so we could do:
sstables::compaction_descriptor({sst}, 0, std::numeric_limits<uint64_t>::max(), utils::make_random_uuid(), sstables::compaction_options::make_reshard());
I think we can postpone this to a followup cleanup patch.

Reply:
@glommer @raphaelsc sure, this is mostly syntactic sugar. let's do that in a follow-up patch then.